packages/shared/src/hooks/usePostById.ts

```diff
@@ -32,9 +32,11 @@ const usePostById = ({
     { ...options, enabled: !!id && tokenRefreshed },
   );
 
+  const post = postById || (options?.initialData as PostData);
+
   return useMemo(
     () => ({
-      post: postById?.post,
+      post: post?.post,
```

Comment:
I think no changes are required here since the `options` was passed and the `postById` will get the value of it on the first render.